bundles/model/org.eclipse.smarthome.model.persistence/src/org/eclipse/smarthome/model/persistence/extensions/PersistenceExtensions.java

```diff
@@ -395,6 +396,47 @@ static public DecimalType averageSince(Item item, AbstractInstant timestamp, Str
         return new DecimalType(average);
     }
 
+    /**
+     * Gets the sum of the state of a given <code>item</code> since a certain point in time. 
+     * The default persistence service is used. 
+     * 
+     * @param item the item to get the average state value for
+     * @param the point in time to start the check 
+     * @return the average state value since the given point in time
+     */
+    static public DecimalType sumSince(Item item, AbstractInstant timestamp) {
+        if(isDefaultServiceAvailable()) {
+            return sumSince(item, timestamp, defaultService);
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Gets the sum of the state of a given <code>item</code> since a certain point in time. 
+     * The {@link PersistenceService} identified by the <code>serviceName</code> is used. 
+     * 
+     * @param item the item to get the average state value for
+     * @param the point in time to start the check 
+     * @param serviceName the name of the {@link PersistenceService} to use
+     * @return the sum state value since the given point in time
+     */
+
+    static public DecimalType sumSince(Item item, AbstractInstant timestamp, String serviceName) {
+        Iterable<HistoricItem> result = getAllStatesSince(item, timestamp, serviceName);
+        Iterator<HistoricItem> it = result.iterator();
+        
+        double sum = 0;
+        while(it.hasNext()) {
+            State state = it.next().getState();
+            if (state instanceof DecimalType) {
+                sum += ((DecimalType) state).doubleValue();
```

Comment:
Why do you use the (unprecise) double value here instead of summing up the (exact) BigDecimals?